torch/_decomp/decompositions.py

```diff
@@ -1361,6 +1361,54 @@ def addmv(self: Tensor, mat1: Tensor, vec: Tensor, beta: int = 1, alpha: int = 1
     return out + beta * self
 
 
+@register_decomposition(aten.pixel_shuffle)
+@out_wrapper()
+@pw_cast_for_opmath
+def pixel_shuffle(self: Tensor, upscale_factor: int):
+    batch = self.shape[:-3]
+    C_out = self.shape[-3] // upscale_factor**2
+    HW_out = (self.shape[-2] * upscale_factor, self.shape[-1] * upscale_factor)
+    n = len(batch)
+    return (
+        self.view(
+            *batch,
+            C_out,
+            upscale_factor,
+            self.shape[-2],
+            upscale_factor,
+            self.shape[-1],
```

Comment:
This looks like unshuffle? Docs say pixel_shuffle has input shape `*, C * r**2, H, W`

Reply:
I'm not sure I understand the question. Input has shape `*, C_out * r**2, H, W` and output has shape `*, C_out, H * r, W * r`

Reply:
You are viewing `self` as `*, C, r, H, r, W` which doesn't make sense?

Reply:
Yeah, it's a view  of `*, C * r**2 H, W` as `*, C, r, H, r, W` and then some transposes to get the `output`.

Reply:
Surely this should be `*, C, r, r, H, W`?

Reply:
Which is also what the comments in the ATen kernel state
https://github.com/pytorch/pytorch/blob/9c1348feb3de872f7cabd807abbc228e7192cd46/aten/src/ATen/native/cpu/PixelShuffleKernel.cpp#L39

Reply:
In case it isn't clear, say we have a contiguous tensor
```
size = (C * r * r, H, W), stride = (H*W, W, 1)
```
your view will result in
```
size = (C, r, H, r, W), stride = (r*r*H*W, r*H*W, r*W, W, 1)
```
where the stride `r * W` is meaningless. No dimension of the input tensor has that stride.

The actual strides we want are `(r * r * H * W, r * H * W, H * W, W)` as you can see from the kernel (in slightly different notation),
https://github.com/pytorch/pytorch/blob/9c1348feb3de872f7cabd807abbc228e7192cd46/aten/src/ATen/native/cpu/PixelShuffleKernel.cpp#L32-L37

Reply:
Thanks for elaborating. Sorry for misunderstanding  your comments. I've fixed the issue.